test/jpypetest/test_fault.py

```diff
@@ -0,0 +1,1249 @@
+import _jpype
+import jpype
+from jpype import *
+import common
+
+
+class FaultTestCase(common.JPypeTestCase):
+    """ Test for fault paths in JPype
+
+    This test is only executed if fault instrumentation is compiled in.
+    Fault instrumentation is trigger as part of the coverage compilation.
+
+    This test suite brutally tries to force an exception to be thrown
+    at each entry point and function call.  The exception is controlled
+    based on the name of the function in the JP_TRACE_IN and JP_PY_TRY
+    block.  Specific fault points are also triggered to produce
+    abnormal objects which can then be passed to trigger error handling
+    behaviors for off normal conditions.
+
+    This may be one of the most tedious files every written by human hands.
+    Coffee is advised.
+
+    """
+
+    def setUp(self):
+        common.JPypeTestCase.setUp(self)
+
+    @common.requireInstrumentation
```

Comment:
I think this decorator could also be attached at class definition level

Reply:
My long term plan is to move these tests from the giant fault handler test back to the unit test for each of the corresponding portions.  I did that for the primitives.  So the decorator was used to make the tests more portable.

Reply:
agreed